cmd/reject/main.go

```
@@ -0,0 +1,121 @@
+	flag.StringVar(&googleDocID, "google-doc-id", "", "reject a single doc (optional)")
+	flag.Parse()
+
+	cfg := config.MustLoadConfig()
+	logger := config.SetupLogger()
+
+	// Signal handling
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+
+	rejectService := initRejectService(cfg, logger)
+	rejectService.Config = specs.RejectConfig{
+		DryRun: dryRun,
+	}
+
+	// Handle single file testing
+	if googleDocID != "" {
+		logger.Info("testing with single file", "file_id", googleDocID)
+		if err := rejectService.RejectSpecByGoogleDocID(ctx, googleDocID); err != nil {
+			logger.Error("failed to reject spec", "error", err)
+			os.Exit(1)
+		}
+		logger.Info("single file test completed")
+		return
+	}
+
+	// Set up ticker for periodic runs
+	ticker := time.NewTicker(cfg.GetRejectInterval())
+	defer ticker.Stop()
+
+	logger.Info("starting rejection job",
+		"interval", cfg.GetRejectInterval(),
+		"dry_run", dryRun,
+		"pid", os.Getpid())
+
+	// Run initial rejection
+	if err := rejectService.RejectAllStaleSpecs(ctx); err != nil {
+		logger.Error("spec rejection failed", "error", err)
+	}
+	// Exit after initial dry run: no need to continue running on loop
+	if dryRun {
+		return
+	}
+
+	// Wait for either context cancellation or ticker
+	for {
+		select {
+		case <-ctx.Done():
+			logger.Info("rejection job stopped")
+			return
+		case <-ticker.C:
+			logger.Info("starting periodic rejection iteration")
+			if err := rejectService.RejectAllStaleSpecs(ctx); err != nil {
+				logger.Error("spec rejection failed", "error", err)
+			}
+		}
+	}
+}
+
+// initRejectService initializes the Reject service with all its dependencies
+func initRejectService(cfg *config.Config, logger *slog.Logger) *specs.RejectService {
+	dbConn, err := db.NewDB(logger, cfg)
+	if err != nil {
+		logger.Error("failed to connect to database", "error", err.Error())
+		os.Exit(1)
+	}
```

Nitpick: instead of calling logger and exit here, you could just return an error:
```go
return fmt.Errorf("failed to connect to database: %v", err)
```

Then your `main` function becomes the only point in which your program can exit. This makes it easier to reason about, as you only have one function that can exit your program:
```go
rejectService, err := initRejectService(cfg, logger)
if err != nil {
    logger.Error("cannot initialize reject service", "error", err)
	os.Exit(1)
}
```